pkg/ottl/functions.go

```diff
@@ -412,9 +412,9 @@ func (o Optional[T]) get() reflect.Value {
 
 // Allows creating an Optional with a value already populated for use in testing
 // OTTL functions.
-func NewTestingOptional[T any](val T) Optional[T] {
+func NewTestingOptional[T any](val T, hasValue bool) Optional[T] {
```

Comment:
I'm open to other opinions on whether this is how we want to do it, but I don't think we need to change this. For Optional arguments with a value, we can use `NewTestingOptional(val)` and for Optional arguments without a value we can use `Optional{}`, which is how it would look if the user didn't pass it in an invocation.

Reply:
Yes, this approach looks more realistic than explicitly setting `hasValue` to  `false`.